scripts/coins-logon-widget.js

```diff
@@ -239,74 +255,76 @@ CoinsLogonWidget.prototype.onError = function(error) {
+        this.update({ usernameProps: clearErrorProps });
+    }
 
-    this.form.setToLogout(statusText);
+    if (event.target.name === 'password') {
+        this.update({ passwordProps: clearErrorProps });
+    }
 };
 
-CoinsLogonWidget.prototype.onLogout = function(response) {
-    this.form.setToLogin();
+CoinsLogonWidget.prototype.onLogin = function(response) {
+    // Redirect if the option is set and there's a URL
+    if (this._options.redirect && this._options.redirectUrl) {
+        window.location = this._options.redirectUrl;
+    } else {
+        this.update({
+            isLoggedIn: true,
+            username: response.username,
+        });
+    }
 };
 
-CoinsLogonWidget.prototype.onSubmit = function(event) {
-    event.preventDefault();
+CoinsLogonWidget.prototype.onLogout = function() {
+    this.update({ isLoggedIn: false });
+};
 
-    var errors;
+CoinsLogonWidget.prototype.onPasswordExpired = function() {
+    var message = utils.callOrReturn(this._options.messages.passwordExpired);
+    this.update({ errorMessage: message });
+};
 
-    this.formGroups.forEach(function(formGroup) {
-        var isValid = formGroup.validate();
+CoinsLogonWidget.prototype.onPasswordWillExpire = function(response) {
+    var passwordExpiration = Date.parse(response.user.passwordExpDate);
+    var now = Date.now();
+    var message = utils.callOrReturn(
+        this._options.messages.passwordWillExpire,
+        [utils.formatDay((passwordExpiration - now) / CoinsLogonWidget.DAY)]
+    );
+    this.update({ errorMessage: message });
+};
 
-        //TODO: emit validation error events
-        if (!isValid) {
-            errors = true;
-        }
-    });
+/* ==========================================================================
+   Static properties
+   ========================================================================== */
 
-    if (!errors) {
-        this.emit('submitted', event);
+CoinsLogonWidget.DAY = 24 * 60 * 60 * 1000;
 
-        //TODO: Make authentication pluggable
-        this.login();
-    }
+CoinsLogonWidget.EVENTS = {
+    INVALID: 'invalid',
+    LOGIN: 'login:init',
+    LOGIN_ERROR: 'login:error',
+    LOGIN_ACCOUNT_EXPIRED: 'login:accountExpired',
+    LOGIN_ACCOUNT_WILL_EXPIRE: 'login:accountWillExpire',
+    LOGIN_PASSWORD_EXPIRED: 'login:passwordExpired',
+    LOGIN_PASSWORD_WILL_EXPIRE: 'login:passwordWillExpire',
+    LOGIN_SUCCESS: 'login:success',
+    LOGOUT: 'logout:init',
+    LOGOUT_ERROR: 'logout:error',
+    LOGOUT_SUCCESS: 'logout:success'
 };
 
 CoinsLogonWidget.VERSION = __VERSION__; // Injected via Webpack
 
-CoinsLogonWidget.DEFAULTS = {
-    classNames: {
-        root: 'coins-logon-widget'
-    },
-    formGroups: [{
-        inputName: 'username',
-        labelText: 'Username:',
-        placeholder: ''
-    }, {
-        inputName: 'password',
-        labelText: 'Password:',
-        placeholder: '',
-        type: 'password'
-    }],
-    hiddenLabels: false,
-    horizontal: false,
```

Comment:
Bye, bye, meaningless defaults. Although I just remembered the `horizontal` argument isn’t honored anymore. Crap.